src/migrate/index.js

```diff
@@ -108,9 +115,49 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.boolean('is_locked');
+    });
+  }
+
+  _getLockTableName(tableName) {
+    return tableName + '_lock';
+  }
+
+  _isLocked() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .count('is_locked as count')
+      .then(function(data) {
+        if (data[0].count > 0) {
+          return true;
+        }
+        return false;
+      });
+  }
+
+  _lockMigrations() {
```

Comment:
@elhigu I updated the method of locking to use a single record which gets updated but I’m unclear what you mean by using forUpdate in this circumstance. My understanding is that the row will already be locked while it get’s updated. But maybe I misunderstood something.